odbc_fdw.c

```diff
@@ -174,7 +177,62 @@ static struct odbcFdwOption valid_options[] =
 	{ NULL,       InvalidOid}
 };
 
-typedef enum { TEXT_CONVERSION, HEX_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+typedef enum { TEXT_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+
+static GetDataTruncation
+result_truncation(SQLRETURN ret, SQLHSTMT stmt)
+{
+    SQLCHAR sqlstate[5];
+    GetDataTruncation truncation = NO_TRUNCATION;
+    if (ret == SQL_SUCCESS_WITH_INFO)
+    {
+		SQLGetDiagRec(SQL_HANDLE_STMT, stmt, 1, sqlstate, NULL, NULL, 0, NULL);
+        if (strcmp((char*)sqlstate, ODBC_SQLSTATE_STRING_TRUNCATION) == 0)
```

Comment:
Maybe use `strncmp` instead of `strcmp` for more safety.

Reply:
Agreed (although you already spoiled the fun of buffer-overflowing in your previous comment, bah)